src/main/kotlin/com/expedia/graphql/schema/extensions/annotationExtensions.kt

```diff
@@ -69,32 +69,38 @@ internal fun KAnnotatedElement.isGraphQLID() = this.findAnnotation<GraphQLID>()
 internal fun Annotation.getDirectiveInfo(): DirectiveAnnotation? =
     this.annotationClass.annotations.find { it is DirectiveAnnotation } as? DirectiveAnnotation
 
-internal fun KAnnotatedElement.directives() =
+internal fun KAnnotatedElement.directives(config: SchemaGeneratorConfig) =
     this.annotations.asSequence()
-        .mapNotNull { it.getDirectiveInfo() }
-        .map { it.getGraphQLDirective() }
+        .mapNotNull { it.getGraphQLDirective(config) }
         .toList()
 
 @Throws(CouldNotGetNameOfAnnotationException::class)
-private fun DirectiveAnnotation.getGraphQLDirective(): GraphQLDirective {
-    val kClass: KClass<out DirectiveAnnotation> = this.annotationClass
+private fun Annotation.getGraphQLDirective(config: SchemaGeneratorConfig): GraphQLDirective? {
```

Comment:
Instead of building directives based just on `GraphQLDirectives` (aka `DirectiveAnnotation` here) we also include the base annotation, e.g.:

```kotlin
@GraphQLDirective(description = "A permission the user needs to hold to use this") // <---- getDirectiveInfo()
annotation class HasPermission(val permission: Permission) // <---- `this`
```